http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -344,15 +410,52 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
     )(res)
   }
 
-  protected def saveClusterError(runtimeId: Long, errorMessage: String, errorCode: Int, now: Instant): F[Unit] =
-    dbRef
-      .inTransaction {
-        clusterErrorQuery.save(runtimeId, RuntimeError(errorMessage, errorCode, now))
-      }
-      .void
-      .adaptError {
-        case e => new Exception(s"Error persisting cluster error with message '${errorMessage}' to database: ${e}", e)
+  protected def saveRuntimeError(runtimeId: Long,
+                                 errorDetails: Option[RuntimeErrorDetails],
+                                 startUpScriptOutputFile: Option[GcsPath])(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val result = for {
+      ctx <- ev.ask
+      errorToPersist <- errorDetails match {
+        case Some(error) =>
+          F.pure(error)
+        case None =>
+          startUpScriptOutputFile match {
```

Comment:
Moved this logic from `DataprocRuntimeMonitor`.. though it seems that `startUpScriptOutputFile` was always passed as `None` anyway